### ISSUE TITLE
[5.1.0 BETA 3] Fist fight and shield fight LRs anticheats are bugged

### ISSUE THREAD

**Issue by azalty** [azalty/sm-hosties-plus#9]
### This bug has been found on version 5.1.0 BETA 3.

**Describe the bug**
Attacking the enemy with the correct weapon (fists or shield) causes the attacker to be slain by the anticheat.

French chat logs:
```
// Fist fight
[Ba_Jail]  Player1 a sélectionné Combat à mains nues contre Player2.
[Ba_Jail]  Player1 a été tué pour avoir attaqué Player2 avec fists durant une DV.
[Ba_Jail]  Player2 won the Combat à mains nues!

// Shield fight
[Ba_Jail]  Player1 a sélectionné Bataille de bouclier contre Player2.
[Ba_Jail]  Player2 a été tué pour avoir attaqué Player1 avec shield durant une DV.
[Ba_Jail]  Player1 won the Bataille de bouclier!
```

**Error logs**
No error logs.

**Comment by azalty** [comment]
Currently investigating as I didn't find any logic reason for this to happen.

I've added logging in 5.1.0 BETA 4 to try to understand why this happens.

**Comment by azalty** [comment]
Fixed in 5.1.0 BETA 6 ([commit](https://github.com/azalty/sm-hosties-plus/commit/0e7413c1b4f5c6e78d298d6d6cc41506cef83359)) by applying the weapon field fix for OnTakeDamage to all games instead of CS:S only. Damage must now be done directly in order for the fix to work to prevent bugs.

**Comment by azalty** [comment]
The fix is out on the stable version 5.1.0
